colossus/src/main/scala/colossus/protocols/http/Header.scala

```diff
@@ -62,37 +64,96 @@ object HttpHeaders {
   val CookieHeader      = "cookie"
   val SetCookie         = "set-cookie"
   val TransferEncoding  = "transfer-encoding"
+
+  def apply(hdrs: HttpHeader*) : HttpHeaders = new HttpHeaders(hdrs.toArray)
+}
+
+trait HttpHeader {
+  def key: String
+  def value: String
+  def encoded: Array[Byte]
+
+  override def equals(that: Any): Boolean = that match {
+    case that: HttpHeader => this.key == that.key && this.value == that.value
+    case other => false
+  }
+
+  override def hashCode = (key + value).hashCode
+
+  override def toString = s"($key,$value)"
+
+}
+
+//generally created when encoding http responses
+class EncodedHeader(val encoded: Array[Byte], keyLength: Int, valueStart: Int) extends HttpHeader {
+  lazy val key = new String(encoded.take(keyLength))
+  lazy val value = new String(encoded.drop(valueStart))
 }
 
-//TODO: support for pulling values as types other than String
-trait HttpHeaderUtils {
+//generally created when parsing http requests
+class DecodedHeader(val key: String, val value: String) extends HttpHeader {
+  lazy val encoded = (key + ": " + value).getBytes("UTF-8")
 
-  def headers : Seq[(String, String)]
+  def toEncodedHeader = new EncodedHeader(encoded, key.length, key.length + 2)
+}
+
+
+object HttpHeader {
+  def apply(key: String, value: String): HttpHeader = (new DecodedHeader(key, value)).toEncodedHeader
 
 
+  object Conversions {
```

Comment:
I like this idea.  Does it make sense to look into one that is a conversion for (String, Any), which just invokes .toString on Any?